### PR TITLE
WIP - Miecd Vietnam

### PR DESCRIPTION
**Pull request by HusnaHariz**
Web app for MIECD-Vietnam project

### CODE REVIEW

File: src/store/ducks/sms_events.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  };
+  //   console.log("gets called", actionCreated);
+  return actionCreated;
+};
+
+// The reducer
+
+/** interface for sms state in redux store */
+// tslint:disable-next-line: class-name
+interface smsStateReducer {
```
Comment (p-netm) [inline src/store/ducks/sms_events.ts]: Use PascalCase for type names

File: src/store/ducks/sms_events.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+/** the sms reducer function */
+export default function reducer(
+  state: smsStateReducer = initialState,
+  action: TestActionTypes
+): smsStateReducer {
+  switch (action.type) {
+    case FETCHED_SMS:
+      return {
+        ...state,
+        payload: action.payload,
```
Comment (p-netm) [inline src/store/ducks/sms_events.ts]: see this issue openSRP/opensrp-web#24 for context on the problem here, the issue is fixed in   openSRP/opensrp-web#31. The specific fix of interest is in this [commit](https://github.com/OpenSRP/opensrp-web/pull/31/commits/929e05e00f072701cb68d88872149ea9809819de)

File: src/components/RiskColoring/index.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+const riskColoring = (props: RiskColoringProps) => {
+  const { Risk } = props;
+  switch (Risk) {
+    case 'No Risk':
+      return (
+        <td className="green" id="default-width">
+          {Risk}
+        </td>
+      );
+    case 'no risk':
```
Comment (p-netm) [inline src/components/RiskColoring/index.tsx]: the code for the first 2 cases is quite indetical, maybe a better approach would be to standardize the the Risk prop, say change it to lowerCase or uppercase.

or have a look up table to decide only those parts that are unique and then return a single jsx instance; something like:

```typescript
  /** Interface for the risk look up dictionary Object */
  interface ClassNameAndTextLookup {
    [key: string]: { className: string; label: string };
  }

  // standardize the Risk variable's string by changing case
  const upperCasedrisk = Risk.toUpperCase();

  /** lookup dictionary for different Risk categories */
  const classNameAndTextLookup: ClassNameAndTextLookup = {
    HIGH: { className: 'yellow', label: 'High risk' },
    LOW: { className: 'green', label: 'Low Risk' },
    'NO RISK': { className: 'green', label: 'No Risk' },
    RED: { className: 'red', label: 'Red Alert' },
    DEFAULT: { className: 'risk-default', label: Risk },
  };

  // use default if given Risk definition was not found in our dictionary
  const risk = classNameAndTextLookup.upperCasedRisk ? upperCasedrisk : 'DEFAULT';

  return (
    <td className={classNameAndTextLookup[risk].className} id="default-width">
      {classNameAndTextLookup[risk].label}
    </td>
  );
}
```

File: src/containers/LogFace/index.tsx [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    supersetFetch('1277').then((result: any) => {
+      fetchTestDataActionCreator(result);
+    });
+  }
+
+  // tslint:disable-next-line: no-empty
+  public handleSubmit() {}
+  // tslint:disable-next-line: no-empty
+  public handleTermChange() {}
+  public handleKeyPress(event: { key: string }) {
```
Comment (p-netm) [inline src/containers/LogFace/index.tsx]: React has types for DomEvents.

File: src/configs/settings.ts [hunk 5]
```diff
@@ -43,4 +43,13 @@ if (ENABLE_ONADATA_OAUTH) {
   };
 }
 
+export const Types = [
```
Comment (moshthepitt) [inline src/configs/settings.ts]: Can you give this a better name please?  What 'Types' are these?  Keep in mind that OpenSRP web will be used in very many ways so all variables need to be as descriptive as possible.

File: src/containers/LogFace/index.tsx [hunk 6]
```diff
@@ -69,41 +94,46 @@ export class LogFace extends React.Component<PropsInterface, {}> {
...
+                        {location}
+                      </DropdownItem>
+                    );
+                  })}
+                </DropdownMenu>
+              </Dropdown>
             </div>
             <div className="location-type-filter">
               Type
-              <Form onSubmit={this.handleSubmit}>
```
Comment (moshthepitt) [inline src/containers/LogFace/index.tsx]: Please use the Formik library like we do everywhere else.

File: src/containers/LogFace/index.tsx [hunk 7]
```diff
@@ -2,59 +2,84 @@
...
-      this.handleSearch();
+  public componentWillReceiveProps(nextProps: any) {
+    if (!this.props.testData.length) {
+      this.setState({
+        filteredData: nextProps.testData,
+      });
     }
   }
+
+  public handleSubmit(e: any) {
```
Comment (moshthepitt) [inline src/containers/LogFace/index.tsx]: If you use something like Formik you dont have to do this.

File: src/containers/pages/Home/Home.tsx [hunk 8]
```diff
@@ -31,7 +33,7 @@ class Home extends React.Component<{}, {}> {
                   <p>Display of all messages receioved from MIECD in chronological order</p>
                 </div>
                 <div id="cont-sized">
-                  <a href="/log-face">
+                  <a href={LOGFACE_URL}>
```
Comment (moshthepitt) [inline src/containers/pages/Home/Home.tsx]: Is it possible to make a prop on the Home component for this?

File: src/components/RiskColoring/index.tsx (hunk 3 above)
Comment (FrankApiyo) [reply src/components/RiskColoring/index.tsx]: I went with making all the risk prop lowercase

**Pull request closed without merging**